### PR TITLE
Feature/gap 1850 spadmin

### PR DESCRIPTION
**Pull request by dylanwrightCO**

### CODE REVIEW

File: src/main/java/gov/cabinetofice/gapuserservice/dto/UserDto.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+import java.util.List;
+
+@Builder
+@Data
+public class UserDto {
+    private String gap_user_id;
+    private String email;
+    private String sub;
+    private List<Role> roles;
```
Comment (dominicwest) [inline src/main/java/gov/cabinetofice/gapuserservice/dto/UserDto.java:16]:
```suggestion
    private List<RoleDto> roles;
```

File: src/main/java/gov/cabinetofice/gapuserservice/repository/UserRepository.java [hunk 2]
```diff
@@ -1,14 +1,22 @@
...
 @Repository
 public interface UserRepository extends JpaRepository<User, Integer> {
+
+    @EntityGraph(attributePaths = {"department", "roles"})
     Optional<User> findByEmail(String email);
+    @EntityGraph(attributePaths = {"department", "roles"})
     Optional<User> findBySub(String sub);
+
+    @EntityGraph(attributePaths = {"department", "roles"})
+    Optional<User> findById(int id);
```
Comment (dominicwest) [inline src/main/java/gov/cabinetofice/gapuserservice/repository/UserRepository.java:19]: Do we defo wanna lazy load if we're gonna eagerly fetch these relations each query anyway?

File: src/main/java/gov/cabinetofice/gapuserservice/security/WebSecurityConfig.java [hunk 3]
```diff
@@ -51,7 +51,10 @@ public WebSecurityCustomizer webSecurityCustomizer() {
                 "/redirect-after-cola-login",
                 "/error/**",
                 "/.well-known/jwks.json",
-                "/logout"
+                "/logout",
+                "/super-admin-dashboard",
+                "/user/**",
+                "/user/**/department"
```
Comment (dominicwest) [inline src/main/java/gov/cabinetofice/gapuserservice/security/WebSecurityConfig.java:57]: We would want to be authenticated for these endpoints

**Review by dominicwest — CHANGES_REQUESTED** [review]

File: src/main/java/gov/cabinetofice/gapuserservice/repository/UserRepository.java (hunk 2 above)
Comment (dylanwrightCO) [reply src/main/java/gov/cabinetofice/gapuserservice/repository/UserRepository.java:19]: I think I'd rather lazy load as a default and then explicitly eagerly fetch.  Means we are not over fetching in future queries by accident.

**Review by dominicwest — APPROVED** [review]
LGTM! Might nest the unit tests for a particular method under a class like in E.g. the UserTest class just for readability, but not that fussed

**Pull request merged**